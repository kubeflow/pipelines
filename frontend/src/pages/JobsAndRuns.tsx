/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import AllRunsList from './AllRunsList';
import JobList from './JobList';
import MD2Tabs from '../atoms/MD2Tabs';
import { Page, PageProps } from './Page';
import { RoutePage } from '../components/Router';
import { classes } from 'typestyle';
import { commonCss, padding } from '../Css';

export enum JobsAndRunsTab {
  JOBS = 0,
  RUNS = 1,
}

export interface JobsAndRunsProps extends PageProps {
  view: JobsAndRunsTab;
}

interface JobAndRunsState {
  selectedTab: JobsAndRunsTab;
}

class JobsAndRuns extends Page<JobsAndRunsProps, JobAndRunsState> {

  public getInitialToolbarState() {
    return { actions: [], breadcrumbs: [] };
  }

  public render() {
    return (
      <div className={classes(commonCss.page, padding(20, 'lr'))}>
        <MD2Tabs tabs={['Group by job', 'Show all runs']} selectedTab={this.props.view}
          onSwitch={this._tabSwitched.bind(this)} />
        {this.props.view === 0 && (
          <JobList {...this.props} />
        )}

        {this.props.view === 1 && (
          <AllRunsList {...this.props} />
        )}
      </div>
    );
  }

  public async load(): Promise<void> {
    return;
  }

  private _tabSwitched(newTab: JobsAndRunsTab) {
    this.props.history.push(newTab === JobsAndRunsTab.JOBS ? RoutePage.JOBS : RoutePage.RUNS);
  }
}

export default JobsAndRuns;
