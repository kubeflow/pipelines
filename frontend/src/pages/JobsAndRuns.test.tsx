/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import JobsAndRuns, { JobsAndRunsTab } from './JobsAndRuns';
import { shallow } from 'enzyme';

describe('JobsAndRuns', () => {
  it('renders jobs page', () => {
    const routerProps = { view: JobsAndRunsTab.JOBS, someOtherProps: 'any' };
    expect(shallow(<JobsAndRuns {...routerProps as any} />)).toMatchSnapshot();
  });

  it('renders runs page', () => {
    const routerProps = { view: JobsAndRunsTab.RUNS, someOtherProps: 'any' };
    expect(shallow(<JobsAndRuns {...routerProps as any} />)).toMatchSnapshot();
  });

  it('switches to clicked page by pushing to history', () => {
    const spy = jest.fn();
    const routerProps = { view: JobsAndRunsTab.JOBS, history: { push: spy } };
    const tree = shallow(<JobsAndRuns {...routerProps as any} />);

    tree.find('MD2Tabs').simulate('switch', 1);
    expect(spy).toHaveBeenCalledWith('/runs');

    tree.find('MD2Tabs').simulate('switch', 0);
    expect(spy).toHaveBeenCalledWith('/jobs');
  });
});
