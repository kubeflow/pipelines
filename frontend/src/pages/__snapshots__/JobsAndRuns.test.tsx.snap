// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsAndRuns renders jobs page 1`] = `
<div
  className="page"
>
  <MD2Tabs
    onSwitch={[Function]}
    selectedTab={0}
    tabs={
      Array [
        "Group by job",
        "Show all runs",
      ]
    }
  />
  <JobList
    history={Object {}}
    location=""
    match=""
    toolbarProps={Object {}}
    updateBanner={[Function]}
    updateDialog={[MockFunction]}
    updateSnackbar={[MockFunction]}
    updateToolbar={[Function]}
    view={0}
  />
</div>
`;

exports[`JobsAndRuns renders runs page 1`] = `
<div
  className="page"
>
  <MD2Tabs
    onSwitch={[Function]}
    selectedTab={1}
    tabs={
      Array [
        "Group by job",
        "Show all runs",
      ]
    }
  />
  <AllRunsList
    history={Object {}}
    location=""
    match=""
    toolbarProps={Object {}}
    updateBanner={[Function]}
    updateDialog={[MockFunction]}
    updateSnackbar={[MockFunction]}
    updateToolbar={[Function]}
    view={1}
  />
</div>
`;
